### PR TITLE
MINOR: Remove usages of `scala.collection.JavaConversions`, and some typo fixes

### PR DESCRIPTION
**Pull request by vahidhashemian**
### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala [hunk 1]
```diff
@@ -535,19 +534,18 @@ class DefaultMessageFormatter extends MessageFormatter {
...
-      valueDeserializer.get.configure(JavaConversions.propertiesAsScalaMap(propertiesWithKeyPrefixStripped("value.deserializer.", props)).asJava, false)
+      valueDeserializer.get.configure(propertiesWithKeyPrefixStripped("value.deserializer.", props).asScala.toMap.asJava, false)
     }
   }
 
   private def propertiesWithKeyPrefixStripped(prefix: String, props: Properties): Properties = {
     val newProps = new Properties()
-    import scala.collection.JavaConversions._
-    for ((key, value) <- props if key.startsWith(prefix) && key.length > prefix.length)
+    for ((key, value) <- props.asScala.toMap if key.startsWith(prefix) && key.length > prefix.length)
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Doesn't seem like calling toMap on every entry is a good idea.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. One comment. Also worth mentioning the typos in the title since that ends up being most of the PR.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala (hunk 1 above)
Comment (vahidhashemian) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Is it on every entry? or once the whole `props` object?
In any case, I refactored it to look more like Scala.

**Review by vahidhashemian — COMMENTED** [review]
@ijuma thanks for the feedback. Updated the title in the new commit.
Not sure if I got your point on the inline comment. Made an update anyway, please take a look. Thanks.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala (hunk 1 above)
Comment (ijuma) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: My bad, I misread the code.

**Review by ijuma — APPROVED** [review]
I pushed a couple of trivial changes. LGTM after that. @vahidhashemian, do they look OK to you too?

**Comment by ijuma** [comment]
Actually, it seems like we still use JavaConversions in InternalTopicIntegrationTest, KafkaEmbedded and EmbeddedKafkaCluster. Any reason why we can't replace those @vahidhashemian?

**Comment by vahidhashemian** [comment]
@ijuma, your updates look good. I had missed those other ones in the Java code. Removed them in the new commit.

**Comment by ijuma** [comment]
Retest this please

**Comment by vahidhashemian** [comment]
I think the failure was related to this [hotfix](https://github.com/apache/kafka/commit/7b0254e00f137c574783341358b654c7a38f6135). I did a rebase.

**Comment by vahidhashemian** [comment]
Retest this please

**Comment by vahidhashemian** [comment]
Sorry, forgot to pull your earlier commit :(

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request merged**